### ISSUE TITLE
Error calling CodeNarcServer via java : Could not reserve enough space for ...

### ISSUE THREAD

**Issue by Sunchock** [nvuillam/vscode-groovy-lint#144]
Hi, i'm trying to get this extension working on Windows and using vscode.
When I tried to lint a groovy file in my IDE, i got : 

```
npm-groovy-lint error: Fatal error while calling CodeNarc
Reason: unknown

undefined
If you still have an error, post an issue to get help: https://github.com/nvuillam/vscode-groovy-lint/issues
```

After more reading, I found this error in the logs:
```
2022-02-03T15:10:32.393Z npm-groovy-lint Error calling CodeNarcServer via java:
{"status":1,"stdout":"Error occurred during initialization of VM\nCould not reserve enough space for 2097152KB object heap\n""stderr":"",
"childJavaProcess":{"_events":{},"_eventsCount":2,"_closesNeeded":3,"_closesGot":3,"connected":false,"signalCode":null,
"exitCode":1,"killed":false,"spawnfile":"java","_handle":null,
"spawnargs":["java","-Xms256m","-Xmx2048m","-cp","\"c:\\Users\\    etc.
```
I tried to search this error on the web and found this : https://stackoverflow.com/questions/4401396/could-not-reserve-enough-space-for-object-heap

But I'm afraid the error is coming directly from the java command arguments in vscode-groovy-lint.
Looks like the same problem here : https://github.com/nvuillam/vscode-groovy-lint/issues/93#issuecomment-874596345
Any idea how to fix it please ?

**Comment by nvuillam** [comment]
CodeNarc (embedded within npm-groovy-lint, its embedded within VsCode GroovyLint) can be very resource consuming... how much RAM is there on your VM ? :/

**Comment by Sunchock** [comment]
> CodeNarc (embedded within npm-groovy-lint, its embedded within VsCode GroovyLint) can be very resource consuming... how much RAM is there on your VM ? :/

I never used CodeNarc before, I not sure about which VM you are talking about. My computer has 16 GB of RAM. I can allocate more RAM to the "VM" if needed.

**Comment by nvuillam** [comment]
@Sunchock what java version are you using ?

**Comment by Sunchock** [comment]
@nvuillam 
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) Client VM (build 25.131-b11, mixed mode)

**Comment by nvuillam** [comment]
@Sunchock this is fixed in newest release v3.0.0, thanks to the great job of @stevenh :)

Please can you confirm it works well for your case now ? (you might need to install java 17 or higher)
- If yes, please rate the app -> https://marketplace.visualstudio.com/items?itemName=NicolasVuillamy.vscode-groovy-lint&ssr=false#review-details
- If not, please reopen the issue and we'll check :)

**Comment by Sunchock** [comment]
> @Sunchock this is fixed in newest release v3.0.0, thanks to the great job of @stevenh :)
> 
> Please can you confirm it works well for your case now ? (you might need to install java 17 or higher)
> 
> * If yes, please rate the app -> https://marketplace.visualstudio.com/items?itemName=NicolasVuillamy.vscode-groovy-lint&ssr=false#review-details
> * If not, please reopen the issue and we'll check :)

I confirm, it works wonderfully now! 😄

**Comment by nvuillam** [comment]
@Sunchock thanks a lot for your feedback :)
